src/irmin-pack/unix/store.ml

```diff
@@ -392,7 +394,7 @@ module Maker (Config : Conf.S) = struct
             Lwt.return_unit
         end
 
-        let is_split_allowed = Gc.is_allowed
+        let is_split_allowed = Gc.is_allowed ~action:`Delete
```

Comment:
This isn't quite right since split should be allowed for stores where GC is allowed when archiving.

To achieve this, you could add something like a `gc_action` argument to pass through to `Gc.is_allowed`. You would also need to add this argument to the `split` function in order to check the appropriate `is_split_allowed`. At first pass, it seems a bit unfortunate to dirty the split api with gc action concerns, but it has the benefit of more clearly communicating the intrinsic connection between split and gc.

Reply:
I was worrying about dirtying the `split` API indeed, but this is probably necessary for now indeed.

Reply:
Though this makes me think: does it make sense for the same store to be gc'ed in archival and deletion mode at different times?

Would it be fair to instead choose the GC action at store opening (or even creation), and use that choice implicitly internally instead of re-exposing the choice on each API call? (at the risk of the user mixing flags by error).

Reply:
Aha, you're right that it doesn't make sense to mix GC calls of archive and delete. I really like your idea of embedding this as a configuration choice... and we already have that in the `lower_root`! We can decide which type of GC to use based on this being set or not:

```ocaml
let action = match lower_root with
| Some _ -> `Archive
| None -> `Delete
```
So we would never have to expose this to the user and thus keep our API nice and tidy. :nail_care: 

In hindsight, keeping the action as a flag on the GC API is really a remnant of the original purpose which was to carry a volume identifier and thus trigger potential creation of a new volume. With the removal of the volume identifier and signalling of volume creation, I think we can go the next step and remove the action entirely!

What do you think?

Reply:
I really like this! We just need to make really clear the implications of setting `lower_root` at store opening :)

Reply:
Nice! Yeah I agree. We should update the docs for the config value to make this clear.

Also, once we have the entire mechanism in place, we should either update the [existing gc example](https://github.com/mirage/irmin/blob/main/examples/irmin-pack/gc.ml) or create a new one that uses the lower layer.